internal/service/url_service.go

```diff
@@ -21,23 +26,76 @@ func NewURLService(repo repository.URLRepository) URLService {
 }
 
 func (s urlService) CreateShortURL(req models.ShortenRequest) (string, error) {
-	//TODO implement me
-	panic("implement me")
+	short := shortener.GeneratorShortURL()
+	log.Println(short)
+	_, err := s.repo.GetURL(short)
+	if err != nil {
+		return "", fmt.Errorf("short url already exists %s ", short)
+	}
+	return short, nil
 }
 
 func (s urlService) GetOriginalURL(shortURL string) (string, error) {
-	//TODO implement me
-	panic("implement me")
+
+	originalUrl, err := s.repo.GetURL(shortURL)
+	if err != nil {
+		return "", fmt.Errorf("short url not found %s", shortURL)
+	}
+	return originalUrl.OriginalURL, nil
+
 }
 
 func (s urlService) LogRedirect(shortURL, referrer string) error {
-	//TODO implement me
-	panic("implement me")
+
+	id := uuid.NewString()
```

Comment:
Don't create new variable if it's used only ones

Reply:
logEntry := models.RedirectLog{
		ID:          uuid.NewString(),
		ShortURL:   shortURL,